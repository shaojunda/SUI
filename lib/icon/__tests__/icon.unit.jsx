import renderer from "react-test-renderer";
import React from "react";
import Icon from "../icon";
import {mount} from 'enzyme';
describe("icon", () => {
  it("render successfully", () => {
    const json = renderer.create(<Icon className="alipay" />).toJSON();
    expect(json).toMatchSnapshot();
  });

  it("onclick", () => {
    let n = 1
    const fn = () => {
      n = 2
    }
    const component = mount(<Icon name='alipay' onClick={fn}></Icon>)
    component.find('svg').simulate('click')
    expect(n).toEqual(2)
  });

  it("onclick second way", () => {
    const fn = jest.fn()
    const component = mount(<Icon name='alipay' onClick={fn}></Icon>)
    component.find('svg').simulate('click')
    expect(fn).toBeCalled()
  });
});
