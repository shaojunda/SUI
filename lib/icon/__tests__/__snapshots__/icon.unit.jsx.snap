// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icon is a svg 1`] = `
<svg
  className="sui-icon alipay"
>
  <use
    xlinkHref="#undefined"
  />
</svg>
`;

exports[`icon render successfully 1`] = `
<svg
  className="sui-icon alipay"
>
  <use
    xlinkHref="#undefined"
  />
</svg>
`;
